specs/features/Inconsistencies.md

```diff
@@ -0,0 +1,126 @@
+  * This can also present a form of schedule risk because if it so happens that the last platform ends up being the hardest, you can end up with more work than calendar
+* Type coercion in Javascript can cause it to be more permissive than other platforms leading to some constructs like `maxlines: "2"` being valid in JS, but invalid in, say, UWP (if JS code isn't carefully written -- we do, of course, author in Typescript, but we do also have to keep these problems in mind)
+* Distinct implementations means that defaults can vary by platform (e.g. whether the `TextBlock` `wrap` property defaults to `true` if not specified)
+* Differing choices made by devs at implementation time can lead to divergent behaviors (e.g. whether an invalid property value is silently dropped vs. emitting a warning vs. failing to parse the card)
+* Supporting libraries and platform affordances available to us can ease or speed up implementation, but can also be the source of inconsistencies (e.g. differences in how a dev constructs .NET's `DateTime`, UWP's `DateTime`, C++'s `std::chrono::time_point`, and Javascript's `Date` -- instantiating each object with, e.g a string can vary in permissivity)
+  * Indeed, we've had a few issues arise from differences between a hand-rolled solution and a pre-rolled solution (e.g. date/time handling (#1915) and support for markdown (#1984))
+
+### Rendering
+
+Rendering presents another avenue by which inconsistencies can sneak into Adaptive Cards. Some of these problems can be pretty tricky to solve.
+
+* Multiple implementations means that "gotcha" scenarios might not always be covered or may be covered inconsistently (e.g. what happens when you combine `maxLines`, markdown lists, and `wrap: true`?)
+* Platform-provided controls can vary widely in their behaviors and support. Even if divergent behavior is caught during implementation, it may be difficult or tricky to solve completely
+* Some 
+
+## Proposals
+
+* Attempt to reduce code duplication:
+  * Experiment with the shared model as Javascript (e.g. via [emscripten](https://emscripten.org/))
+  * Move .NET from using Newtonsoft.JSON to using the shared model via P/Invoke (Note that this will have impacts on partners. It's likely that this move will only make sense if Javascript is also using the shared model)
+* Changes processes to prevent new inconsistencies from ever happening:
+  * New bugs automatically filed with `inconsistency` tag, which is only removed once it's determined that the issue is *not* a consistency issue
+  * We can't ship if an `inconsistency`-tagged bug hasn't been resolved for this release
+  * Features that need implementation across all platforms are merged to a single branch and PR'd/tested for inconsistencies before being allowed to merge upstream
```

Comment:
i like this idea. can we also pick a desinated designer (or an architect) per the feature? and require the chosen designer to present design to the design meeting with minimum of a prototype and brief overview of the design. After the review approval, the designer implement the feature on at least one platform and complete document on how the feature works. Then other devs pick up the work on the remaining platforms. Designers can be anyone. 
This approach will save dev times by not having to design features each time working on a different platform. Although platforms are different, our feature is common across the all platforms. Feature implementation design doesn't need to be re-desined, and this also allow the plaform specialzation to benefit us. If we are tight on schedule, we can assign feature developement on iOS to the dev with iOS specialization. If we have leeway on schedule, give to dev who wants to learn about a platform. 

Sometimes a plaform can be too different that the design can not be applied directly, for exaple, if negative margin is used in .NET, but iOS can't use negative margin. Even such case, the implementor doesn't need to start from scratch, and the design can serve as scafolding for the dev's own implementation. This also gives opportunity to document. Documenting the difference is much easier than writing a whole new document.

it's aslo benetfitial in maintanence perspective. Since we have several implementors of a feature, and if the feature works the same way, we could assign any devs to work on any platforms (yes, this is a little bit of exaggeration). There will be differences, but the consistent feature implementation will serve as bread crumbs to navigate through differences in platforms, and it will save time in debugging.